### PR TITLE
Fix transforming named function expressions into generators

### PR DESCRIPTION
**Pull request by UltCombo**
Bug report: https://github.com/codecombat/codecombat/issues/2874#issuecomment-120497951

### CODE REVIEW

File: test/es6_spec.coffee [hunk 1]
```diff
@@ -329,6 +329,28 @@ describe "JavaScript Test Suite", ->
...
+          this.hesitate();
+          this.hesitate();
+          this.hesitate();
+        };
+        this.f();
+        this.slay();
+      """
+      aether.transpile code
+      f = aether.createFunction()
+      gen = f.apply dude
```
Comment (UltCombo) [inline test/es6_spec.coffee]: I'm not sure why `f` is undefined here. Am I doing something wrong?
The fix works fine inside CodeCombat.

Comment (UltCombo) [inline test/es6_spec.coffee]: Oh, it is because I've fixed `makeYieldAutomatically` only and this test uses `makeYieldConditionally`. Gawd I'm dumb.

**Comment by UltCombo** [comment]
Fixed an issue and deleted obsolete comments.
The other 13 test failures are unrelated to this change.

**Comment by UltCombo** [comment]
@trotod FYI

**Comment by nwinter** [comment]
Awesome, thanks for fixing that one! How are you finding the Aether codebase?

**Comment by UltCombo** [comment]
Well, I'm still finding my way around it, but a few things I noted:

- Some algorithms could do with more comments.

- Looks like the `grunt-jasmine-node` dependency is pretty outdated, which in turn means the [`jasmine-node`](https://github.com/mhevery/jasmine-node) version is also very outdated. I couldn't find a way to focus on a specific unit test (e.g. using up-to-date node-jasmine's [`iit` function](https://github.com/mhevery/jasmine-node#jasmine)), and so I had to run the entire test suite when I was only interested in a single test.

- The Traceur dependency is very outdated as well, though this doesn't seem to be an immediate problem.

- (personal note) I'm still not used to reading CoffeeScript syntax, so it takes a higher cognitive effort to understand the code.

- The project structure is very well organized, which is a big plus.

**Comment by nwinter** [comment]
Hmm, I thought someone hooked up a way to run just single tests earlier, but I can't find it; guess not. Would be happy to have grunt-jasmine-node updated. Traceur, we can't update without figuring out the new way to hook into its compilation system, which I had given up trying to do slightly after the version we are now using, since it was in crazy flux; maybe it's more stable now. I don't know what we would gain from an update. Ideally we can figure out a way to engineer around it, since it generates a huge amount of code for the generators. Regenerator doesn't work because it's far slower to compile, unfortunately.

**Pull request merged**